src/OpenCLScheduler.h

```diff
@@ -36,22 +37,16 @@ class OpenCLScheduler {
                  std::vector<net_t>& output_pol,
                  std::vector<net_t>& output_val);
 private:
-    class ForwardTask {
-    public:
-        const std::vector<net_t> *input;
-        std::vector<net_t> * output;
-        std::promise<void> prom;
-        ForwardTask() : input(nullptr), output(nullptr) {}
-        ForwardTask(const std::vector<net_t> * in,
-                    std::vector<net_t> * out)
-            : input(in), output(out) {}
-    };
-
     std::vector<std::unique_ptr<OpenCL_Network>> m_networks;
     std::vector<std::unique_ptr<OpenCL>> m_opencl;
-    Utils::ThreadPool m_threadpool;
-};
 
-extern OpenCLScheduler opencl;
+    // XXX : I failed to figure out how to make this compile with unique_ptr
+    // especially on visual studio
+    using OpenCLContextQueue = std::list<std::shared_ptr<OpenCLContext>>;
+    std::vector<OpenCLContextQueue> m_context;
```

Comment:
Really a list of contexts per worker thread. m_work_threads? m_thread_contexts? Something else? Not sure.

Reply:
Well, it's the 'pool' for OpenCL contexts.  Hence I just put it m_context_pool.  Not perfect but I don't think we should call this 'threads' anymore to avoid confusion (although yes it still has a lot of sense of the concept 'thread' here)